README.md

```diff
@@ -1,27 +1,305 @@
 # Descope SDK for Python
 
-Use the Descope SDK for Python to quickly and easily add user authentication to your application or website.
+The Descope SDK for python provides convenient access to the Descope user management and authentication API
+for a backend written in python. You can read more on the [Descope Website](https://descope.com).
 
-The SDK supports Python 3.6 and above.
+## Requirements
 
-## Installing the SDK
+The SDK supports Python 3.7 and above.
 
-Replace any instance of `<ProjectID>` in the code below with your company's Project ID, which can be found in the [Descope console](https://app.descope.com).
+## Installing the SDK
 
-Run the following code in your project. These commands will add the Descope ExpresSDK for Python as a project dependency, and set the `DESCOPE_PROJECT_ID` variable to a valid \<ProjectID\>.
+Install the package with:
 
-```code
+```bash
 pip install descope
-export DESCOPE_PROJECT_ID=<ProjectID>
 ```
 
-## What do you want to implement?
+## Setup
+
+A Descope `Project ID` is required to initialize the SDK. Find it on the
+[project page in the Descope Console](https://app.descope.com/settings/project).
+
+```python
+from descope import DescopeClient
+
+# Initialized after setting the DESCOPE_PROJECT_ID env var
+descope_client = DescopeClient()
+
+# ** Or directely **
+descope_client = DescopeClient(project_id="<Project ID>")
+```
+
+## Usage
+
+Here are some examples how to manage and authenticate users:
+
+### OTP Authentication
+
+Send a user a one-time password (OTP) using your preferred delivery method (_email, SMS, Whatsapp message_). An email address or phone number must be provided accordingly.
+
+The user can either `sign up`, `sign in` or `sign up or in`
+
+```python
+from descope import DeliveryMethod
+
+# Every user must have an identifier. All other user information is optional
+email = "desmond@descope.com"
+user = {"name": "Desmond Copeland", "phone": "212-555-1234", "email": email}
+descope_client.otp.sign_up(method=DeliveryMethod.EMAIL, identifier=email, user=user)
+```
+
+The user will receive a code using the selected delivery method. Verify that code using:
+
+```python
+jwt_response = descope_client.otp.verify_code(
+    method=DeliveryMethod.EMAIL, identifier=email, code=value
+)
+```
+
+The session and refresh JWTs should be returned to the caller, and passed with every request in the session. Read more on [session validation](#session-validation)
```

Comment:
maybe some example, on how to extract from jwt_response, the actual jwt, fields and refresh token, that will tie it better IMO to session validation section